pyam/core.py

```diff
@@ -485,7 +487,9 @@ def append(self, other, ignore_meta_conflict=False, inplace=False,
         ret.meta = merge_meta(ret.meta, other.meta, ignore_meta_conflict)
 
         # append other.data (verify integrity for no duplicates)
-        _data = ret._data.append(other._data, verify_integrity=True)
+        _data = ret._data.append(other._data)
```

Comment:
I guess you have to be careful here to not change `df._data` (if operating inplace) and then raising the error afterwards

Reply:
No you don't.

Pandas is very clear on when it operates in place and when it does not. The only inplace operations are:
1. __setitem__ (`df.loc[..] = ...`, `df[...] = ...`, ...) 
2. __setattr__  (`df.index = ...`, `df.name = ...`)
3. methods where you add an explicit `inplace=True` argument.

All other methods operate on *shallow* copies and there are non-inplace replacements for all of them, ie instead
of `df["test"] = "bla"`, you can use `df.assign(test="bla")`, instead of `df.index = ind`, you can use `df.set_axis(ind)`, and so on and not operate inplace at all. If one follows this behaviour systematically, one can actually avoid most of the deep copies, pyam is currently relying on.

Long story short, here we are not updating `ret._data` only `_data`.

Reply:
Ok, got it, thanks for the clarification!